### PR TITLE
Make entity tables pageable

### PR DESCRIPTION
**Pull request by dnlkoch**
This enables server side paging (including sorting) in the entity tables.

Requires [!412](https://github.com/terrestris/shogun-util/pull/412).

Note: The filter is still client sided as there is not backend support available right now.

Peek 2023-07-13 15-28

Please review @terrestris/devs.

### CODE REVIEW

File: src/Component/FormField/UserPermissionGrid/UserPermissionModal/UserPermissionModal.tsx [hunk 1]
```diff
@@ -73,7 +73,7 @@ const UserPermissionModal: React.FC<UserPermissionModalProps> = ({
       setUsersLoading(true);
 
       try {
-        setUsers(await client.user().findAll());
+        setUsers((await client.user().findAll()).content);
```
Comment (KaiVolland) [inline src/Component/FormField/UserPermissionGrid/UserPermissionModal/UserPermissionModal.tsx:76]: Splitting this up would enhance readability.

File: src/Component/GeneralEntity/GeneralEntityRoot/GeneralEntityRoot.tsx [hunk 2]
```diff
@@ -706,6 +740,27 @@ export function GeneralEntityRoot<T extends BaseEntity>({
           loading={isGridLoading}
           size="small"
           tableConfig={tableConfig}
+          // @ts-ignore
```
Comment (KaiVolland) [inline src/Component/GeneralEntity/GeneralEntityRoot/GeneralEntityRoot.tsx:743]: Why is this needed?

File: src/Component/ImageFile/ImageFileTable/ImageFileTable.tsx [hunk 3]
```diff
@@ -158,14 +191,35 @@ export const ImageFileTable: React.FC<ImageFileTableProps> = ({
...
-      pagination={false}
       onRow={(record) => {
         return {
           onClick: () => onRowClick(record)
         };
       }}
       bordered
       size="small"
+      loading={loadingState === 'loading'}
+      // @ts-ignore
```
Comment (KaiVolland) [inline src/Component/ImageFile/ImageFileTable/ImageFileTable.tsx:202]: see above

**Review by KaiVolland — APPROVED** [review]
Nice. Just some minor none blocking notes.

File: src/Component/Dashboard/DashboardStatistics/DashboardStatistics.tsx [hunk 4]
```diff
@@ -38,7 +38,7 @@ export const DashboardStatistics: React.FC<DashboardStatisticsProps> = ({
     setLoadingState('loading');
     try {
       const fetchedEntities = await service?.findAll();
-      const count = fetchedEntities.length || 0;
+      const count = fetchedEntities.content.length || 0;
```
Comment (LukasLohoff) [inline src/Component/Dashboard/DashboardStatistics/DashboardStatistics.tsx:41]: minor: the content size should also be available via `content.totalElements`

File: src/Component/GeneralEntity/GeneralEntityRoot/GeneralEntityRoot.tsx [hunk 5]
```diff
@@ -142,6 +148,11 @@ export function GeneralEntityRoot<T extends BaseEntity>({
   const [isFormLoading, setFormLoading] = useState<boolean>(false);
   const [isSaving, setIsSaving] = useState<boolean>(false);
   const [isUploadingFile, setIsUploadingFile] = useState<boolean>(false);
+  const [pageTotal, setPageTotal] = useState<number>();
+  const [pageSize, setPageSize] = useState<number>(10);
```
Comment (LukasLohoff) [inline src/Component/GeneralEntity/GeneralEntityRoot/GeneralEntityRoot.tsx:152]: we could make this configurable in `admin-client-config.js`

File: src/Component/ImageFile/ImageFileTable/ImageFileTable.tsx [hunk 6]
```diff
@@ -18,36 +24,43 @@ import config from 'shogunApplicationConfig';
...
+}];
 
 export const ImageFileTable: React.FC<ImageFileTableProps> = ({
   ...passThroughProps
 }) => {
 
+  const [entities, setEntities] = useState<ImageFile[]>();
+  const [loadingState, setLoadingState] = useState<'failed' | 'loading' | 'done'>();
+  const [pageTotal, setPageTotal] = useState<number>();
+  const [pageSize, setPageSize] = useState<number>(10);
```
Comment (LukasLohoff) [inline src/Component/ImageFile/ImageFileTable/ImageFileTable.tsx:59]: same here

File: src/Component/GeneralEntity/GeneralEntityRoot/GeneralEntityRoot.tsx (hunk 5 above)
Comment (LukasLohoff) [reply src/Component/GeneralEntity/GeneralEntityRoot/GeneralEntityRoot.tsx:152]: or is this complicated because it may be linked to the styling, i.e. size of the table container?

**Review by LukasLohoff — APPROVED** [review]

File: src/Component/GeneralEntity/GeneralEntityRoot/GeneralEntityRoot.tsx (hunk 2 above)
Comment (dnlkoch) [reply src/Component/GeneralEntity/GeneralEntityRoot/GeneralEntityRoot.tsx:743]: I'll have a look at it again, but I fear the typing in antd is not correct.

File: src/Component/GeneralEntity/GeneralEntityRoot/GeneralEntityRoot.tsx (hunk 5 above)
Comment (dnlkoch) [reply src/Component/GeneralEntity/GeneralEntityRoot/GeneralEntityRoot.tsx:152]: Good idea, will check that!

File: src/Component/GeneralEntity/GeneralEntityRoot/GeneralEntityRoot.tsx (hunk 2 above)
Comment (dnlkoch) [reply src/Component/GeneralEntity/GeneralEntityRoot/GeneralEntityRoot.tsx]: See 45255f0.

Not sure why, but the component extended the `HTMLDivElement` type and this interfered with the `onChange` type of the table.

File: src/Component/Dashboard/DashboardStatistics/DashboardStatistics.tsx (hunk 4 above)
Comment (dnlkoch) [reply src/Component/Dashboard/DashboardStatistics/DashboardStatistics.tsx:41]: Sure, fixed in 5ebd4b9.

File: src/Component/GeneralEntity/GeneralEntityRoot/GeneralEntityRoot.tsx (hunk 5 above)
Comment (dnlkoch) [reply src/Component/GeneralEntity/GeneralEntityRoot/GeneralEntityRoot.tsx]: See cd1c80e.

**Comment by dnlkoch** [comment]
Thanks for your reviews!
